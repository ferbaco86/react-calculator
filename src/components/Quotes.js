import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import NavBar from './NavBar';
import data from '../quotes.json';

const QuoteTitle = styled.h1`
font-size: 2.6rem;
text-align: center;
padding: 4.5rem;
margin-top: 10rem;
color: white;`;

const Quotes = () => {
  const randomNumber = Math.round(Math.random() * 20);
  const [quote, setQuote] = useState('Here comes the Math Quotes');
  useEffect(() => {
    setQuote(data[randomNumber]);
  }, []);
  return (
    <>
      <NavBar />
      <QuoteTitle data-testid="quote">{quote}</QuoteTitle>
    </>
  );
};

export default Quotes;
