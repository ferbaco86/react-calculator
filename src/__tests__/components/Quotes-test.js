import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import {
  render, screen,
} from '@testing-library/react';
import Quotes from '../../components/Quotes';

it('renders correctly', () => {
  const tree = renderer.create(<BrowserRouter><Quotes /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Quotes Components', () => {
  test('Have a quote', () => {
    render(<BrowserRouter><Quotes /></BrowserRouter>);
    const quote = screen.getByText(content => content.startsWith('Quote:'));
    expect(quote).toBeInTheDocument();
  });
});
