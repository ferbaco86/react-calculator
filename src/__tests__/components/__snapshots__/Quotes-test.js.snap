// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="sc-bdfBwQ eOkbzq"
  >
    <h1
      className="sc-gsTCUz joJsAy"
    >
      Math Magicians
    </h1>
    <div
      className="sc-dlfnbm XLyPB"
    >
      <a
        className="sc-hKgILt iUuVZI"
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        className="sc-hKgILt iUuVZI"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        className="sc-hKgILt iUuVZI"
        href="/quotes"
        onClick={[Function]}
      >
        Quotes 
      </a>
    </div>
  </nav>,
  <h1
    className="sc-eCssSg bwjHmh"
    data-testid="quote"
  >
    Here comes the Math Quotes
  </h1>,
]
`;
